vowpalwabbit/feature_group.h

```diff
@@ -277,7 +277,48 @@ struct features
   }
 
   // if one wants to add proper destructor for features, make sure to update ezexample_predict::~ezexample_predict();
-  // ~features() { ... }
+  ~features() { 
+     values.delete_v();
+     indicies.delete_v();
+     space_names.delete_v();
+   }
+   features(const features&) = delete;
+   features & operator=( const features& ) = delete;
+   
+   
+   // custom move operators required since we need to leave the old value in
+   // a null state to prevent freeing of shallow copied v_arrays
+   features(features&& other) :
+   values(std::move(other.values)),
+   indicies(std::move(other.indicies)),
+   space_names(std::move(other.space_names)),
+   sum_feat_sq(other.sum_feat_sq)
+   {
+     // We need to null out all the v_arrays to prevent double freeing during moves
+     auto & v = other.values;
+     v._begin = v._end = v.end_array = nullptr;
```

Comment:
I'd prefer if we did not chain the assignments like this. We will get warnings for this too

Reply:
Replace with v_init to zero out?